shark-graph/src/main/java/shark/internal/ByteSubArray.kt

```diff
@@ -5,21 +5,21 @@ package shark.internal
  */
 internal class ByteSubArray(
   private val array: ByteArray,
-  private val range: IntRange,
+  private val rangeStart: Int,
+  private val rangeEndInclusive: Int,
   private val longIdentifiers: Boolean
 ) {
-  private val endInclusive = range.endInclusive - range.start
```

Comment:
I'm a bit confused about indexes here, probably I'm understanding it wrong. Here's my thought process:

Originally we created range as `valueIndex until valueIndex + bytesPerValue`, so `endInclusive` was set to `range.endIncusive - range.start = valueIndex + bytesPerValue - valueIndex = bytesPerValue`

With new implementation we pass `rangeEndInclusive` as `(valueIndex + bytesPerValue) - 1` and use it as a replacement for `endInclusive`. 

But, obviously, `(valueIndex + bytesPerValue) - 1 != bytesPerValue`. Is it me misunderstanding the flow and `IntRange` here or is there a bug?